### PR TITLE
Add lineNumber, variableId & nodeType in audit sources report

### PR DESCRIPTION
**Pull request by pawan-privado**
This PR adds the functionality for recording line numbers when generating audit sources and modifies existing tests accordingly.

### CODE REVIEW

**Review by dbMundada — APPROVED** [review]
LGTM

File: src/main/scala/ai/privado/audit/DataElementDiscovery.scala [hunk 1]
```diff
@@ -248,38 +268,54 @@ object DataElementDiscovery {
         case (key, value) => {
           val isCollectionInput = if (collectionInputList.contains(key.fullName)) "YES" else "NO"
           if (taggedMemberInfo.contains(key.fullName)) {
+            val path   = key.file.name.headOption.getOrElse(Constants.EMPTY)
+            val tpe    = AuditReportConstants.ELEMENT_DISCOVERY_NODE_TYPE_METHOD
```
Comment (dbMundada) [inline src/main/scala/ai/privado/audit/DataElementDiscovery.scala:272]: is tpe constant here?

Comment (dbMundada) [reply src/main/scala/ai/privado/audit/DataElementDiscovery.scala:272]: correct this to member

File: src/main/scala/ai/privado/audit/AuditReportEntryPoint.scala [hunk 2]
```diff
@@ -25,7 +24,10 @@ object AuditReportEntryPoint {
     sourceRuleId: String,
     inputToCollection: Boolean,
     collectionEndpointPath: String,
-    collectionMethodFullName: String
+    collectionMethodFullName: String,
+    variableDeclarationLineNumber: String,
+    variableIdentifier: String,
```
Comment (dbMundada) [inline src/main/scala/ai/privado/audit/AuditReportEntryPoint.scala:29]: rename variableIdentifier -> memberId
rename variableType -> nodeType

**Review by dbMundada — APPROVED** [review]

**Review by pandurangpatil — APPROVED** [review]

**Comment by pandurangpatil** [comment]
@dbMundada looks good to me. We can merge it once unit test action is done. I have scheduled it to rerun it.

**Pull request merged**